execution-engine/blessed-contracts/pos/src/stakes.rs

```diff
@@ -105,6 +119,9 @@ impl Stakes {
             .or_insert(*amount);
     }
 
+    // Mateusz: Why is there `validate_bonding` but there's no `validate_unbonding`?
+    // `bond` always succeeds (I guess because it's called only in case `validate_bonding` succeeds,
+    // but `unbond` can return an error. Maybe we should consider being consistent here.
```

Comment:
We want to do all possible checks at the point where the request is made.
In the case of bonding, that means we do the checks when inserting the request into the queue, even though the actual bonding operation (adding to the stakes map) takes place later, in `step`.
In the case of unbonding, we remove from the stakes map immediately, so the checks and the unbonding operation happen in the same contract call.
But if you prefer, I can split the unbonding one in two. They'll just be called right after each other.

Reply:
Right, didn't catch that.